ml-agents/mlagents/training_analytics_side_channel.py

```diff
@@ -0,0 +1,34 @@
+import uuid
+from mlagents_envs.exception import UnityCommunicationException
+from mlagents_envs.side_channel import SideChannel, IncomingMessage, OutgoingMessage
+
+from mlagents.trainers.settings import TrainerSettings
+
+
+class TrainingAnalyticsSideChannel(SideChannel):
+    """
+    Side channel that sends information about the training to the Unity environment so it can be logged.
+    """
+
+    def __init__(self) -> None:
+        # >>> uuid.uuid5(uuid.NAMESPACE_URL, "com.unity.ml-agents/TrainingAnalyticsSideChannel")
+        # UUID('b664a4a9-d86f-5a5f-95cb-e8353a7e8356')
+        super().__init__(uuid.UUID("b664a4a9-d86f-5a5f-95cb-e8353a7e8356"))
+
+    def on_message_received(self, msg: IncomingMessage) -> None:
+        raise UnityCommunicationException(
+            "The TrainingAnalyticsSideChannel received a message from Unity, "
+            + "this should not have happend."
```

Comment:
```suggestion
            + "this should not have happened."
```

Reply:
Whoops, copied from here
https://github.com/Unity-Technologies/ml-agents/blob/c69ad6bec8bacdea94b3627936c2fb9ec8817f49/ml-agents-envs/mlagents_envs/side_channel/engine_configuration_channel.py#L56
(and there's the same thing in EnvironmentParametersChannel)